custom_components/bond/light.py

```diff
@@ -20,8 +28,16 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     #     return
 
     # Add devices
-    add_entities(BondLight(bond, deviceId) for deviceId in bond.getDeviceIds())
+    for deviceId in bond.getDeviceIds():
+        newBondLight = BondLight(bond, deviceId)
 
+        # If the device type is not Ceiling Fan, or it is Ceiling Fan but has no action for light control
+        # then don't create a light instance
+        if newBondLight._properties['type'] == BOND_DEVICE_TYPE_CEILING_FAN and \
+              ( BOND_DEVICE_ACTION_TURNLIGHTON in newBondLight._properties['actions'] or \
+                BOND_DEVICE_ACTION_TURNLIGHTOFF in newBondLight._properties['actions'] or \
+                BOND_DEVICE_ACTION_TOGGLELIGHT in newBondLight._properties['actions'] ):
```

Comment:
I suggest you only check for the presence of one of the light actions. Since other device types may have lights (Fireplace for example. Or even a generic device created via the API)